javalib/src/main/scala/java/io/CharArrayWriter.scala

```diff
@@ -0,0 +1,108 @@
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+*/
+
+package java.io
+
+object CharArrayWriter {
+  private val TOKEN_NULL = "null"
+}
+
+// Adapted from Apache Harmony CharArrayWriter.java
+//   https://github.com/apache/harmony/blob/02970cb7227a335edd2c8457ebdde0195a735733/classlib/modules/luni/src/main/java/java/io/CharArrayWriter.java
+class CharArrayWriter extends Writer {
+  import CharArrayWriter.TOKEN_NULL
+
+  protected var buf: Array[Char] = new Array[Char](32)
+  protected var count: Int = 0
+
+  def this(initialSize: Int) {
+    this()
+    if (initialSize < 0) throw new IllegalArgumentException("size must be >\\= 0")
+    buf = new Array[Char](initialSize)
+  }
+
+  override def close(): Unit = ()
+
+  private def expand(i: Int): Unit = {
+    /* Can the buffer handle @i more chars, if not expand it */
+    if (this.count + i <= this.buf.length) return
+    val newLen = Math.max(2 * this.buf.length, this.count + i)
+    val newbuf = new Array[Char](newLen)
+    System.arraycopy(this.buf, 0, newbuf, 0, this.count)
+    this.buf = newbuf
+  }
+
+  override def flush(): Unit = ()
+
+  def reset(): Unit = this.count = 0
+
+  def size: Int = this.count
+
+  def toCharArray: Array[Char] = {
+    val result = new Array[Char](this.count)
+    System.arraycopy(buf, 0, result, 0, this.count)
+    result
+  }
+
+  override def toString: String = new String(this.buf, 0, this.count)
+
+  override def write(c: Array[Char], offset: Int, len: Int): Unit = {
+    // avoid int overflow
+    if (offset < 0 || offset > c.length || len < 0 || len > c.length - offset) throw new IndexOutOfBoundsException
+    expand(len)
+    System.arraycopy(c, offset, this.buf, this.count, len)
+    this.count += len
+
+  }
+
+  override def write(oneChar: Int): Unit = {
+    expand(1)
+    this.count += 1
+    this.buf(this.count - 1) = oneChar.toChar
+  }
+
+  override def write(str: String, offset: Int, len: Int): Unit = {
+    if (str == null) throw new NullPointerException("buffer is null")
+    if (offset < 0 || offset > str.length || len < 0 || len > str.length - offset) throw new StringIndexOutOfBoundsException
+    expand(len)
+    str.getChars(offset, offset + len, this.buf, this.count)
+    this.count += len
+  }
+
+  def writeTo(out: Writer): Unit = out.write(this.buf, 0, count)
+
+  override def append(c: Char): CharArrayWriter = {
+    write(c)
+    this
+  }
+
+  override def append(csq: CharSequence): CharArrayWriter = {
+    if (null == csq) append(TOKEN_NULL, 0, TOKEN_NULL.length)
+    else append(csq, 0, csq.length)
+
+    this
+  }
+
+  override def append(csq: CharSequence, start: Int, `end`: Int): CharArrayWriter = {
```

Comment:
`end` is not a keyword in Scala so the backticks are not necessary.

Reply:
Updated!